packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart

```diff
@@ -0,0 +1,216 @@
+import 'dart:io';
+
+import 'package:crypto/crypto.dart';
+import 'package:mason_logger/mason_logger.dart';
+import 'package:shorebird_cli/src/code_push_client_wrapper.dart';
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/config/shorebird_yaml.dart';
+import 'package:shorebird_cli/src/doctor.dart';
+import 'package:shorebird_cli/src/formatters/file_size_formatter.dart';
+import 'package:shorebird_cli/src/ios.dart';
+import 'package:shorebird_cli/src/logger.dart';
+import 'package:shorebird_cli/src/platform.dart';
+import 'package:shorebird_cli/src/shorebird_artifact_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_build_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_config_mixin.dart';
+import 'package:shorebird_cli/src/shorebird_environment.dart';
+import 'package:shorebird_cli/src/shorebird_validation_mixin.dart';
+import 'package:shorebird_code_push_client/shorebird_code_push_client.dart';
+
+class PatchIosFrameworkCommand extends ShorebirdCommand
+    with
+        ShorebirdConfigMixin,
+        ShorebirdValidationMixin,
+        ShorebirdBuildMixin,
+        ShorebirdArtifactMixin {
+  PatchIosFrameworkCommand({
+    HashFunction? hashFn,
+  }) : _hashFn = hashFn ?? ((m) => sha256.convert(m).toString()) {
+    argParser
+      ..addOption(
+        'release-version',
+        help: '''
+The version of the associated release (e.g. "1.0.0"). This should be the version
+of the iOS app that is using this module.''',
+        mandatory: true,
+      )
+      ..addFlag(
+        'force',
+        abbr: 'f',
+        help: 'Patch without confirmation if there are no errors.',
+        negatable: false,
+      )
+      ..addFlag(
+        'dry-run',
+        abbr: 'n',
+        negatable: false,
+        help: 'Validate but do not upload the patch.',
+      );
+  }
+
+  final HashFunction _hashFn;
+
+  @override
+  String get name => 'ios-framework-alpha';
+
+  @override
+  String get description =>
+      'Publish new patches for a specific iOS framework release to Shorebird.';
+
+  @override
+  Future<int> run() async {
+    if (!platform.isMacOS) {
```

Comment:
minor but might be cleaner to add this to `validatePreconditions` like:

```dart
await validatePreconditions(
  requiredPlatforms: [Platform.macOS],
  ...
);
```

Reply:
Done.